app/tasks/mailchimp.py

```diff
@@ -0,0 +1,123 @@
+from __future__ import \
+    unicode_literals, \
+    absolute_import
+
+from . import celery
+from flask import current_app
+from mailchimp3 import MailChimp
+from app.api.suppliers import remove
+from app.models import Supplier, SupplierFramework, SupplierDomain, User, Framework
+from app import db
+from itertools import groupby
+from operator import itemgetter
+from requests.utils import default_headers
+from requests.exceptions import RequestException
+from sqlalchemy.orm.util import aliased
+from sqlalchemy import true
+from sqlalchemy.exc import SQLAlchemyError
+from os import getenv
+
+
+class MailChimpConfigException(Exception):
+    """Raised when the MailChimp config is invalid."""
+
+
+def get_client():
+    headers = default_headers()
+    headers['User-Agent'] = 'Digital Marketplace (marketplace.service.gov.au)'
+    client = MailChimp(
+        getenv('MAILCHIMP_USERNAME'),
+        getenv('MAILCHIMP_SECRET_API_KEY'),
+        timeout=10.0,
+        headers=headers
+    )
+    return client
+
+
+@celery.task
+def sync_mailchimp_seller_list():
+    client = get_client()
+    list_id = getenv('MAILCHIMP_SELLER_LIST_ID')
+
+    if not list_id:
+        raise MailChimpConfigException('Failed to get MAILCHIMP_SELLER_LIST_ID from the environment variables.')
+
+    # get the mailchimp list's existing members
+    try:
+        current_members = client.lists.members.all(
+            list_id,
+            fields='members.email_address,members.id',
+            get_all=True
+        )
+    except RequestException as e:
+        current_app.logger.error("An Mailchimp API error occurred, aborting: %s", e)
+        raise e
+
+    current_member_addresses = []
+    try:
+        for member in current_members['members']:
+            current_member_addresses.append(member['email_address'].lower())
+    except KeyError:
+        pass
+
+    # get the addresses from DM service supplier users
+    try:
+        s_alias = aliased(Supplier)
+        statement = User.query\
+            .with_entities(User.email_address)\
+            .distinct(User.email_address)\
+            .filter(
+                User.role == 'supplier',
+                User.active == true(),
+                Framework.slug == 'digital-marketplace',
+                Supplier.status != 'deleted'
+            )\
+            .join(Supplier)\
+            .join(SupplierFramework, Framework)\
+            .join(SupplierDomain, s_alias)
+        results = statement.all()
+    except SQLAlchemyError as e:
```

Comment:
this `try/except` seems redundant.  if for some reason this query blows up it will end up in the logs anyway